e2e-version/npm-to-docker-image.js

```diff
@@ -94,23 +94,98 @@ function convertToNameSearchParam(build) {
   return target.slice(0, target.length - 1);
 }
 
-function httpGet(url) {
+function httpGet(url, maxRetries = 3, retryDelay = 1000) {
   return new Promise((resolve, reject) => {
-    https
-      .get(url, (res) => {
+    let attempts = 0;
+    let timeoutId = null;
+
+    const makeRequest = () => {
+      attempts++;
+      
+      const req = https.get(url, { timeout: 10000 }, (res) => {
         let data = [];
 
         res.on('data', (chunk) => {
           data.push(chunk);
         });
 
         res.on('end', () => {
+          if (timeoutId) {
+            clearTimeout(timeoutId);
+            timeoutId = null;
+          }
+
           const buffer = Buffer.concat(data).toString();
-          resolve(JSON.parse(buffer));
+          
+          // Check if response is successful
+          if (res.statusCode >= 200 && res.statusCode < 300) {
+            try {
+              const jsonData = JSON.parse(buffer);
+              resolve(jsonData);
+            } catch (parseError) {
+              const error = new Error(`Failed to parse JSON response from ${url}: ${parseError.message}`);
+              error.responseBody = buffer.substring(0, 500); // Include first 500 chars for debugging
+              
+              if (attempts < maxRetries) {
+                console.warn(`JSON parse error on attempt ${attempts}/${maxRetries} for ${url}. Retrying in ${retryDelay * attempts}ms...`);
```

Comment:
Can we use `core.warning` instead of console.warn? That way it will appear as an annotation in the action ui.